controllers/authController.js

```diff
@@ -0,0 +1,24 @@
+const User = require('../models/user');
+const bcrypt = require("bcryptjs");
+
+exports.login = async(req, res, next) => {
+  const email = req.body.email;
+  const password = req.body.password;
+
+  let user = await User.findOne({email});
+
+  if( user && await bcrypt.compare(password, user.password) ) {
+    req.session.logged = true;
+    res.render('rosterview/selectRoster',{
+      viewTitle: 'Select a date',
+      item: user.fname,
+      user_id: user.id
+    });
+  } else {
+    req.session.logged = false;
+    res.render("rosterview/login", {
+      viewTitle: "Login",
+      errors: "Email & Password don't match!"
+    });
+  }
+}
```

Comment:
So here is the corresponding controller for auth and the login method. A couple of changes here first being re-rendering the login template on unauthorized with an extra context variable or errors. There are lots of ways you can do this before you were rendering a JSON error response, which is fine if you're handling the login through ajax. But seeing as you were using handlebars and submitting via the browser seems reasonable and simpler just to let handlebars do the heavy lifting and re-render the login page with errors present.

One thing to note I haven't handled the async error, you can do as before in a try-catch and render a 500 error template page, or you can wrap the login method in a function like `catchAsyncError(fn) => { fn(req, res, next).catch(next) }` and then handle as you see fit.